### PR TITLE
Code mistakes found with TscanCode

### PR DESCRIPTION
**Pull request by HybridDog**
I executed [TscanCode](https://github.com/Tencent/TscanCode) on SuperTux' src folder and it found these errors: [tscan_result.txt](https://github.com/SuperTux/supertux/files/5736634/tscan_result.txt)

I tried to fix them but I don't understand all code sections, so I don't know if all my changes make sense.

### CODE REVIEW

**Review by Semphriss — COMMENTED** [review]
Looked into it, there are some minor details and information about the fixed errors - nothing critical though. I'm responsible for half of them and I'll probably correct them in a cleaner way, to keep my features as straightforward as possible.

File: src/object/player.cpp [hunk 1]
```diff
@@ -1531,9 +1531,6 @@ if (!m_swimming && m_water_jump) {
 		else if ((m_physic.get_velocity_y() <= 0)) {
 		m_sprite->set_action(sa_prefix+"-jump"+sa_postfix);
 		}
-		else if ((m_physic.get_velocity_y() <= 0)) {
-		m_sprite->set_action(sa_prefix+"-skid"+sa_postfix);
-		}
     }
```
[the remark this answers is not in the document]
Comment (HybridDog) [reply src/object/player.cpp:1534]: The skidding action is also set at another place in the code and skidding worked in the game when I tried it.

File: src/sprite/sprite.cpp [hunk 2]
```diff
@@ -79,8 +79,7 @@ Sprite::set_action(const std::string& name, int loops)
   // we prefer that over the parameter.
   m_animation_loops = newaction->has_custom_loops ? newaction->loops : loops;
 
-  // Only reset the frames if both actions don't have the same family name
-  if (m_action->family_name != newaction->family_name)
+  if (!m_action || m_action->family_name != newaction->family_name)
```
[the remark this answers is not in the document]
Comment (HybridDog) [reply src/sprite/sprite.cpp:82]: In the constructor m_action is set to some action from the SpriteData reference m_data, and the draw method only works if m_action is not nullptr. I don't know if a Sprite which cannot be drawn can exist.

**Comment by Semphriss** [comment]
Question - Why "involves: security"? Are there security-critical exploits in the bugs found?

**Comment by HybridDog** [comment]
I think since SuperTux is a singleplayer game there are inherently almost no security problems. Exceptions are the add-on downloading, downloaded script execution (or similar things) and availability problems (e.g. crash on startup when a configuration is wrong; I'm not sure if this is safety or security).
Apparently the security label was never used before: https://github.com/SuperTux/supertux/pulls?q=label%3Ainvolves%3Asecurity+
Here I thought the label may be relevant because some detected errors are about invalid pointers, but it's not critical, so I'm removing it.

**Comment by tobbi** [comment]
Can you please rebase? I'll merge it afterwards.

**Comment by Semphriss** [comment]
@tobbi **Don't merge it yet** - I believe those static variables in the UI might cause a memory leak. I'd like to fix those myself.

@HybridDog Do I have your permission to edit the two static variables?

**Comment by HybridDog** [comment]
> Do I have your permission to edit the two static variables?

Yes, please do it.

I think my changes in game_session.cpp may not make much sense. I don't know in which cases m_currentsector can be null.

**Comment by Semphriss** [comment]
Forgot to write it yesterday, but it's ready now :+1:

**Comment by tobbi** [comment]
Thanks, @HybridDog

**Pull request merged**